sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java

```diff
@@ -408,14 +420,29 @@ private static OrderBy analyzeOrderBy(SelectAnalysis selectAnalysis,
         return new OrderBy(symbols, reverseFlags, nullsFirst);
     }
 
+    private static Symbol analyzeExpression(Expression expression,
+                                            ExpressionAnalyzer selectAnalysisAwareExpressionAnalyzer,
+                                            ExpressionAnalyzer expressionAnalyzer,
+                                            ExpressionAnalysisContext expressionAnalysisContext) {
+        Symbol symbol;
+        if(expression instanceof QualifiedNameReference) {
```

Comment:
why is this distinction necessary here? Isn't it possible to always use the selectAnalysisAwareExpressionAnalyzer.

Seems like this condition here also wouldn't work for functions?